### ISSUE TITLE
Implementation

### ISSUE THREAD

**Issue by bayesianbrad** [bradleygramhansen/pyfo#4]
Implement a correct state that returns the log_pdf, gradients, converts state dict to floats. Converts float values back to Variables with requires_grad= True.
